### DISCUSSION TITLE
Exporting to MP4

### DISCUSSION THREAD

**Discussion by kingcosmo** [scottlamb/moonfire-nvr#140] [Q&A]
Hello, 

I've been running moonfire nvr at home for a week or so (love it and see huge potential!) and had an event I need to investigate.  The videos from this timestamp do not appear in the UI, but I do see files with the relevant timestamp window in the samples directory.  I've copied off these files for inspection, but they do not appear to be valid MP4.  It sounds like from reading the docs that they are stored without requisite headers.  Are there any suggestions for exporting these raw files to video that I can review in VLC? I'm not incredibly familiar with MP4 file format yet, but happy to try.

Best regards,
Cosmo

**Comment by scottlamb** [comment]
The [HTTP API](https://github.com/scottlamb/moonfire-nvr/blob/master/design/api.md#get-apicamerasuuidstreamviewmp4) (used by the UI) is the only supported way to make a proper `.mp4` file.

*   If you don't see the recordings in the UI, my first guess is your clock was off when Moonfire NVR started, and so the timestamps are wrong. [to be precise: the timestamps in the database/API/UI don't match the ones you see in the filesystem.] See [Incorrect timestamps](https://github.com/scottlamb/moonfire-nvr/blob/master/guide/troubleshooting.md#incorrect-timestamps) in the troubleshooting guide.
*   Even if you don't see them in the UI, if you know the filenames from looking around in the same file directory, you can construct a URL for those filenames. The filename is the stream id as eight hexadecimal digits + the recording id as eight hexadecimal digits. That maps to `/api/<camera_uuid>/<stream_type>/view.mp4?s=<recording_id>`. You can look up the camera uuid and stream type for a given stream id in the API responses. (There's also a `moonfire-nvr sql` for directly inspecting through the sqlite3 shell for debugging; you can also look them up that way.)
